src/utils/NavigationUtils.ts

```diff
@@ -14,12 +14,16 @@ import {filterNavigationStackArgs, ScreenRoutesEnum} from '../types';
  */
 export const filterNavigationStack = (args: filterNavigationStackArgs): void => {
```

Comment:
we are going to remove this hacky code with the state machines :P. it was a decent solution at the time. but these state machines will solve this mess

Reply:
Yeah, not sure whether my PR introduced my IDE complaining, or whether it was always present, But added some code and an ignore to stop it from complaining.

Next up removal of it altogether ;)